src/pj_observation.c

```diff
@@ -0,0 +1,91 @@
+/******************************************************************************
+ * Project:  PROJ.4
+ * Purpose:  Implement some service routines for the OBSERVATION generic
+ *           geodetic data type
+ * Author:   Thomas Knudsen,  thokn@sdfe.dk,  2016-06-09/2016-10-11
+ *
+ ******************************************************************************
+ * Copyright (c) 2016, Thomas Knudsen/SDFE
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *****************************************************************************/
+#include <proj.h>
+#include <projects.h>
+
+
+/* Constructor for the OBSERVATION object */
+OBSERVATION pj_observation (
+	double x, double y, double z, double t,
+	double o, double p, double k,
+	int id, unsigned int flags
+) {
+    OBSERVATION g;
+    g.coo.xyzt.x = x;
+    g.coo.xyzt.y = y;
+    g.coo.xyzt.z = z;
+    g.coo.xyzt.t = t;
+    g.anc.opk.o  = o;
+    g.anc.opk.p  = p;
+    g.anc.opk.k  = k;
+    g.id         = id;
+    g.flags      = flags;
+	return g;
+}
+
+
+/* Direction: "+1" forward, "-1" reverse, 0: roundtrip precision check (not fully done) */
+OBSERVATION pj_apply (PJ *P, int direction, OBSERVATION obs) {
```

Comment:
I bet I'll have to look up the meaning og +1, -1 and 0 everytime I use it. Declaring constants like PJ_FORWARD, PJ_REVERSE and PJ_ROUNDTRIP (or something similar) will make calls to pj_apply() self-documenting.

Challenge: Come up with a self-documenting function name that isn't already use :) pj_transform would be a perfect name for this, but is already used... Anyone?